src/Resolver/TemplateMapResolver.php

```diff
@@ -78,51 +84,54 @@ public function setMap($map)
     /**
      * Add an entry to the map
      *
-     * @param  string|array|Traversable $nameOrMap
-     * @param  null|string $path
+     * @param string|iterable<string, string> $nameOrMap
+     * @param null|string $path
      * @throws Exception\InvalidArgumentException
-     * @return TemplateMapResolver
+     * @return $this
      */
     public function add($nameOrMap, $path = null)
     {
-        if (is_array($nameOrMap) || $nameOrMap instanceof Traversable) {
-            $this->merge($nameOrMap);
+        if (is_string($nameOrMap) && ($path === null || $path === '')) {
+            trigger_error(
+                'Using add() to remove individual templates is deprecated and will be removed in version 3.0',
+                E_USER_DEPRECATED,
+            );
```

Comment:
No roadmap has yet been written down for this, but when these deprecation errors are triggered then this should also be reflected in the documentation:

- Migration
  - Preparing for Version 3